docs/wallet/wallet-get-address.md

```diff
@@ -15,3 +15,9 @@ not ever received funds.
 Third, there is `getAddress`. This function takes in an `AccountDb`, `HDChainType`,
 and an `Int`. This will return the address associated with the pubkey at
 the resulting `BIP32Path`.
+
+## Address Queue
+
+The Bitcoin-S wallet uses a background thread meant to ensure safety when fetching addresses.
+This is to ensure independent calls to getNewAddress don't result in a race condition to the database that would generate the same address and cause an error.
+With this background thread, we poll the `addressRequestQueue` seeing if there are any elements in it, if there are, we process them and complete the Promise in the queue.
```

Comment:
I think the only thing I would add here is that parameters are configurable, specifically `bitcoin-s.wallet.addressQueueSize` and `bitcoin-s.wallet.addressQueueTimeout`. 

I would also link them to our example configuration to see how you can configure it: 

https://bitcoin-s.org/docs/next/config/configuration#example-configuration-file